fecgov/openFEC
# Fix PAC and Parties total endpoints

## Summary (required)

- Resolves [#4444 ](https://github.com/fecgov/openFEC/issues/4444)

_rename column sub_id to idx in backend view_
_remove filer: committee_type_
_fix filter: committee_designation_
_add test case_

## How to test the changes locally

-download branch to local, set SQLA_CONN to DEV
-run `pytest`
-run migration 
-test endpoint: ( a tmp view was created on dev for testing purpose )

modify models/totals.py 
`class CommitteeTotalsPacParty(CommitteeTotals):`
    `__tablename__ = 'ofec_totals_pac_party_vw_tmp_hc'`

check these endpoints:
**Before**
https://api.open.fec.gov/v1/totals/pac/?per_page=10&api_key=DEMO_KEY&sort=-receipts&cycle=2020

https://api.open.fec.gov/v1/totals/parties/?per_page=10&api_key=DEMO_KEY&sort=-receipts&cycle=2020

**After change**
http://127.0.0.1:5000/v1/totals/pac/?per_page=10&sort_null_only=false&sort_hide_null=false&sort_nulls_last=false&page=1&sort=-receipts&cycle=2020


http://127.0.0.1:5000/v1/totals/parties/?per_page=10&sort_null_only=false&sort_hide_null=false&sort_nulls_last=false&page=1&sort=-receipts&cycle=2020

committee_designation_filter:
http://127.0.0.1:5000/v1/totals/pac/?sort_nulls_last=false&sort=-cycle&per_page=20&sort_null_only=false&committee_designation=D&cycle=2020&sort_hide_null=false&page=1
